backport_packages/refactor_backport_packages.py

```diff
@@ -0,0 +1,369 @@
+
+
+class RefactorBackportPackages:
+    """
+    Refactors the code of providers, so that it works in 1.10.
+
+    """
+
+    def __init__(self):
+        self.qry = Query()
+
+    def remove_class(self, class_name) -> None:
+        # noinspection PyUnusedLocal
+        def _remover(node: LN, capture: Capture, filename: Filename) -> None:
+            if node.type not in (300, 311):  # remove only definition
+                node.remove()
+
+        self.qry.select_class(class_name).modify(_remover)
+
+    def rename_deprecated_modules(self):
+        changes = [
+            ("airflow.operators.bash", "airflow.operators.bash_operator"),
+            ("airflow.operators.python", "airflow.operators.python_operator"),
+            ("airflow.utils.session", "airflow.utils.db"),
+            (
+                "airflow.providers.cncf.kubernetes.operators.kubernetes_pod",
+                "airflow.contrib.operators.kubernetes_pod_operator"
+            ),
+        ]
+        for new, old in changes:
+            self.qry.select_module(new).rename(old)
+
+    def add_provide_context_to_python_operators(self):
+        # noinspection PyUnusedLocal
+        def add_provide_context_to_python_operator(node: LN, capture: Capture, filename: Filename) -> None:
+            fn_args = capture['function_arguments'][0]
+            fn_args.append_child(Comma())
+
+            provide_context_arg = KeywordArg(Name('provide_context'), Name('True'))
+            provide_context_arg.prefix = fn_args.children[0].prefix
+            fn_args.append_child(provide_context_arg)
+
+        (
+            self.qry.
+            select_function("PythonOperator").
+            is_call().
+            is_filename(include=r"mlengine_operator_utils.py$").
+            modify(add_provide_context_to_python_operator)
+        )
+        (
+            self.qry.
+            select_function("BranchPythonOperator").
+            is_call().
+            is_filename(include=r"example_google_api_to_s3_transfer_advanced.py$").
+            modify(add_provide_context_to_python_operator)
+        )
+
+    def remove_super_init_call(self):
+        # noinspection PyUnusedLocal
+        def remove_super_init_call_modifier(node: LN, capture: Capture, filename: Filename) -> None:
+            for ch in node.post_order():
+                if isinstance(ch, Leaf) and ch.value == "super":
+                    if any(c.value for c in ch.parent.post_order() if isinstance(c, Leaf)):
+                        ch.parent.remove()
+
+        self.qry.select_subclass("BaseHook").modify(remove_super_init_call_modifier)
+
+    def remove_tags(self):
+        # noinspection PyUnusedLocal
+        def remove_tags_modifier(_: LN, capture: Capture, filename: Filename) -> None:
+            for node in capture['function_arguments'][0].post_order():
+                if isinstance(node, Leaf) and node.value == "tags" and node.type == TOKEN.NAME:
```

Comment:
But 1.10.10 does support tags.

(I can't work out from this fn where it is used.)

Reply:
We tested operators on Airflow 1.10.6 and 1.10.10 to encourage more users to migrate.

Reply:
Exactly. We want those operators to also work in previous versions of  Airflow, for those users that cannot migrate.

Reply:
@ashb - also - note this tag removal will only apply to examples, not to the code itself. If you are using the backported operators in Airflow 1.10.10 you can still use tags. I explained that in the comments and examples I added in #8991

Reply:
Just FYI - In the provider packages  - examples are part of the providers and we want the examples to continue being runnable and importable as well for more than airflow 1.10.10. Especially that we are planning to add AIP-4 - System Test automation using those example dags (and so far we are using those example dags to run our manually-triggered system tests, that we run for more versions of Airflow than 1.10.10 .

Reply:
Cool, I did wonder where this might be used, if it's examples only this sound good.

Can you add the reason for doing this change as a comment.

Reply:
Ye. Will do. I have one more thing to fix so I add it there.